Tests/UnitTests/EdgeNetworkServiceTests.swift

```diff
@@ -133,6 +133,29 @@ class EdgeNetworkServiceTests: XCTestCase {
 
         wait(for: [expectation], timeout: 0.5)
     }
+    
+    func testDoRequest_whenConnection_ResponseCode207_CallsCompletionTrue_AndResponseCallback_AndErrorCallback() {
```

Comment:
Can you also add one e2e test with multi status, checking resulting response & error events?